common/predictive-text/message.d.ts

```diff
@@ -257,4 +276,10 @@ interface Suggestion {
    * When suggesting a word, `displayAs` should be that entire word.
    */
   displayAs: string;
+
+  /**
+   * Indicates if this option represents preservation of the user's 
+   * originally-input text (pre-prediction).
+   */
+  isKeep?: boolean;
```

Comment:
Shall we implement the generalized suggestion tag here? e.g.,

```suggestion
  tag?: 'keep';
```

This is to accommodate more suggestion types such as `emoji` and `correction`. Recall [this document](https://docs.google.com/document/d/18_cJA5_rR6ev1aRUteJC7HrHk4LWOTM1oRk8in3fBKY/edit#heading=h.f91dsrwpvozd). I also think clearer names than `keep` may be `verbatim` or `original`

Reply:
Hmm.  That's a pretty reasonable idea, though I'd want to be sure we've thought everything through clearly regarding tags first.  Admittedly, I forgot about the specific document since it's seemed like major back-burner stuff for a future version, so thanks for bringing it up.

* Can a Suggestion have more than one tag?
  - Suppose the text `smile`.
    - `smile` itself is the obvious default 'keep' option.
    - However, 😄 would be a pretty reasonable suggestion, and semantically might be considered a 'keep.'  But yeah, that's probably a 'prediction'.
  - Now consider the text `smilf`.
    - `smile` would be an obvious 'correction'.
    - But what about 😄 here?  It's _both_ an 'emoji' and a 'correction' - do I choose 'emoji' and drop the 'correction' metadata component?
    - If we block 'correction' but allow emojis, we risk the situation where `smilf` does not suggest `smile` (the 'correction') but does suggest 😄 (the 'emoji')... that doesn't seem right.
* I'd prefer to implement the typing here once, so the above point is to ask this - should it be a single string, or would we want an array of tags?

Reply:
A single string. This will make implementation, for us and for developers of custom models, easier. In this specific example, I think the loss of information— that 😄 is a correction of "smilr"— is okay. I assume that emoji would always be the rightmost suggestion, regardless if it is a "correction" or not. I think the "keep" tag should only be emitted when every suggestion the model(s) make suggest something other than the text that should be kept.

Reply:
> I think the "keep" tag should only be emitted when every suggestion the model(s) make suggest something other than the text that should be kept.

I disagree with this part, simply to ensure we can reliably detect keep and preserve it as the first option.  Otherwise, there's a slim chance 'keep' could be an obscure and unlikely word and become buried past the first three suggestions.

Otherwise, 👍 .